### PR TITLE
[sft] support DFT

### PR DESCRIPTION
**Pull request by hjh0119**
https://arxiv.org/abs/2508.05629

https://github.com/modelscope/ms-swift/issues/5338

### CODE REVIEW

File: swift/plugin/loss.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 
 def per_token_loss_func(outputs, labels, **kwargs):
+    enable_dft_loss = kwargs.get('enable_dft_loss', False)
```
Comment (Jintao-Huang) [inline swift/plugin/loss.py]: 这个写到参数表里吧

**Comment by hjh0119** [comment]
# Experiments with DFT vs SFT

training script: [dft.sh](https://github.com/modelscope/ms-swift/blob/main/examples/train/full/dft.sh)

## Training curves  
*(DFT: red, SFT: green)*
training-curves

## Evaluation on MATH500  
Script: https://github.com/yongliang-wu/DFT

| Model (Qwen2.5-Math-1.5B) | MATH500 (%) |
|---------------------------|-------------|
| Origin                    | 31.24       |
| SFT                       | 42.66       |
| DFT                       | **56.81**   |

**Review by tastelikefeet — APPROVED** [review]

**Comment by chuangzhidan** [comment]
then why i get much worse performace (acc) in llama_factory where use_dft_loss: true
